### PR TITLE
Move wpa_supplicant to External

### PR DESCRIPTION
**Pull request by kimusan**
The wpa_supplicant is under a different license and needs to be placed in the external folder
This patch moves the code and modifies it to build in the new location.

### CODE REVIEW

**Comment by sunghan-chang** [comment]
@kimusan Why do you include wpa_supplicant into apps library even though it is in a external folder?

**Comment by kimusan** [comment]
what do you mean? it is removed in the apps folder (files are moved to the external folder).

**Comment by sunghan-chang** [comment]
@kimusan 
There is an existing infrastructure to build packages in external directory.
Your changes in apps directory are NOT acceptable.
If you move it to ext, you should build it using makefile of ext, not apps.

**Comment by kimusan** [comment]
Please check the patch again. it does use the existing make infrastructure in external/ and is build via external/Makefile.  All wpa_supplicant subfolders have seperate Make.defs files as in the other projects in external folder. 
The things changed in apps/ is to get the path to external into apps as the wifi api requires to include some header files from wpa_supplicant (I could have hardcoded the path but I'd rather make it so EXTDIR is propagated dynamically through to the apps  subfolders.

**Comment by kimusan** [comment]
Any news on getting this in?

File: apps/netutils/wifi/Makefile [hunk 1]
```diff
@@ -46,9 +46,9 @@ else
 endif
 endif
 
-CFLAGS += ${shell $(INCDIR) $(INCDIROPT) "$(CC)" $(APPDIR)$(DELIM)wpa_supplicant$(DELIM)src \
-			$(APPDIR)$(DELIM)wpa_supplicant$(DELIM)src$(DELIM)utils \
-			$(APPDIR)$(DELIM)wpa_supplicant$(DELIM)src$(DELIM)common \
+CFLAGS += ${shell $(INCDIR) $(INCDIROPT) "$(CC)" $(TOPDIR)$(DELIM)$(EXTDIR)$(DELIM)wpa_supplicant$(DELIM)src \
```
Comment (sunghan-chang) [inline apps/netutils/wifi/Makefile]: Why do you add wpa_supplicant folders in wifi's makefile?

File: apps/Directory.mk [hunk 2]
```diff
@@ -64,7 +64,7 @@ all: nothing
 
 define SDIR_template
 $(1)_$(2):
-	$(Q) $(MAKE) -C $(1) $(2) TOPDIR="$(TOPDIR)" APPDIR="$(APPDIR)"
+	$(Q) $(MAKE) -C $(1) $(2) TOPDIR="$(TOPDIR)" APPDIR="$(APPDIR)" EXTDIR="$(EXTDIR)"
```
Comment (sunghan-chang) [inline apps/Directory.mk]: How do you get EXTDIR? When makefile of apps is called, EXTDIR is not passed by makefile.unix.

File: apps/netutils/wifi/Makefile (hunk 1 above)
Comment (kimusan) [inline apps/netutils/wifi/Makefile]: wifi api requires some headerfiles only available in that folder.

File: apps/Directory.mk (hunk 2 above)
Comment (kimusan) [inline apps/Directory.mk]: it should do so via this change:
https://github.com/Samsung/TizenRT/pull/278/files#diff-194321126c6a80da39301224fafb4840

Comment (sunghan-chang) [inline apps/Directory.mk]: Oh, I missed. Thank you.

File: external/Makefile [hunk 3]
```diff
@@ -107,7 +122,11 @@ ifeq ($(CONFIG_ENABLE_IOTIVITY),y)
 	$(Q) echo "Launching IoTivity Build"
 	$(Q) TOPDIR="$(TOPDIR)" $(TOPDIR)/../external/iotivity/build_iotivity.sh
 endif
+wpa_supplicant_build:
```
Comment (sunghan-chang) [inline external/Makefile]: Last question.
Why do you add this, unused option?

Comment (kimusan) [inline external/Makefile]: looks to be something I missed during my cleanup of the patch. Dont think it is needed for anything

Comment (sunghan-chang) [inline external/Makefile]: ok. let's remove this.

**Pull request closed without merging**